dsmiddleware/memcache/memcache.go

```diff
@@ -0,0 +1,147 @@
+package memcache
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"github.com/bradfitz/gomemcache/memcache"
+	"go.mercari.io/datastore"
+	"go.mercari.io/datastore/dsmiddleware/storagecache"
+	"time"
+)
+
+var _ storagecache.Storage = &cacheHandler{}
+var _ datastore.Middleware = &cacheHandler{}
+
+const defaultExpiration = 15 * time.Minute
+
+// New memcache middleware creates & returns.
+func New(memcacheAddr string, opts ...CacheOption) interface {
```

Comment:
It might be better to pass in a memcache.Client vs just the address.  This would also be similar to the redis implementation.

Reply:
What will be the benefits of doing that VS just passing the address? My take is that only passing the address requires minimal efforts from the clients side.

Reply:
There's several ways to instantiate the memcache client
https://github.com/bradfitz/gomemcache/blob/master/memcache/memcache.go#L120
With just the address you force one way of using the client.

We made need memcache in other areas of code outside of datastore in which case I think it's preferred to reuse the client.

From a design perspective I think is provides better decoupling and doesn't introduce extra steps the library needs to do.  This would allow the middleware to be "composed" vs instantiated.